src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -30,18 +30,49 @@ static const char* module = "FMI3XML";
 struct fmi3_xml_variable_default_values {
     fmi3_float32_t float32;
     fmi3_float64_t float64;
+    fmi3_uint8_t uint8;
+    fmi3_uint16_t uint16;
+    fmi3_uint32_t uint32;
+    fmi3_uint64_t uint64;
+    fmi3_int8_t int8;
+    fmi3_int16_t int16;
+    fmi3_int32_t int32;
+    fmi3_int64_t int64;
+    fmi3_boolean_t boolean;
+    fmi3_string_t string;
 };
 
-static struct fmi3_xml_variable_default_values VARIABLE_DEFAULT_VALUES = { 0, 0 };
+static struct fmi3_xml_variable_default_values VARIABLE_DEFAULT_VALUES = \
+    { 0, 0, 0, 0, 0, 0, 0, 0, 0, "" };
 
-static void* fmi3_xml_get_type_default_value(fmi3_bitness_enu_t bitness) {
-    switch (bitness) {
-        case fmi3_bitness_32:
-            return &VARIABLE_DEFAULT_VALUES.float32;
-        case fmi3_bitness_64:
+static void* fmi3_xml_get_type_default_value(fmi3_base_type_enu_t baseType) {
```

Comment:
This impl feels like it's too much overhead now. Easier to just use add a field to each baseType and then cast to void*. I mean the whole point was to avoid needing to cast, but it wasn't really a good idea from the start.

Reply:
See if I understood you correctly, it is commit 3e291fc.

Reply:
No, let's discuss it today